### PR TITLE
Fix event option localization variables

### PR DESCRIPTION
**Pull request by XenoAmess**
## Summary

- mirror the native `NEventOptionButton._Ready()` localization order by adding `EventModel.DynamicVars` to each option title and description before formatting
- keep the fix generic: no event IDs or localization variable names are hard-coded in production code
- add regression coverage for missing event-variable formatting and null localization entries

## Live-game evidence

`GameStateService.BuildEventPayload` was repeatedly reaching `LocManager.SmartFormat` from the event-option description getter with only the generic character variables present. Live logs reported missing values including:

- `SUNKEN_STATUE`: `Relic`, `Gold`, `HpLoss`
- `SUNKEN_TREASURY`: `SmallChestGold`, `LargeChestGold`
- `ABYSSAL_BATHS`: `MaxHp`, `Damage`, `Heal`
- `BRAIN_LEECH`: `FromCardChoiceCount`, `CardChoiceCount`, `RipHpLoss`

`SafeReadString` contained each exception but emitted an empty description, so the agent lost the concrete choice effects.

Decompiling the matching game v0.111.0 code confirmed the missing native step: `EventOption` construction adds character details and `IsMultiplayer`, while `NEventOptionButton._Ready()` separately calls `Event.DynamicVars.AddTo(Option.Title)` and `Event.DynamicVars.AddTo(Option.Description)` before `GetFormattedText()`. This PR applies that same ordering in the payload builder.

## Tests

- `dotnet run --project STS2AIAgent.Tests/STS2AIAgent.Tests.csproj --configuration Release` — 50/50 passed
- `dotnet build STS2AIAgent/STS2AIAgent.csproj --configuration Release --property:Sts2DataDir=<game-data-dir>` — passed, 0 warnings / 0 errors

No game deployment or restart was performed for this PR.

## Summary by Sourcery

Include event-specific variables before formatting event option text so localized choices retain their concrete effects.

Bug Fixes:
- Restore event-specific localization variables when formatting event option titles and descriptions, preventing missing-variable errors from producing empty option text.

Enhancements:
- Apply the localization behavior consistently to event payloads and option signatures, including safe handling of null localization entries.

Documentation:
- Document the event option localization failure, its root cause, and the regression coverage.

Tests:
- Add regression tests covering event-variable injection, representative localized event text, signature formatting, and null localization entries.

### CODE REVIEW

File: STS2AIAgent/Game/GameStateService.cs [hunk 1]
```diff
@@ -3913,8 +3913,14 @@ _ when TryGetCombatHandSelection(currentScreen, out var hand) => hand!.CurrentMo
                     {
                         index = i,
                         text_key = SafeReadString(() => opt.TextKey),
-                        title = SafeReadString(() => opt.Title?.GetFormattedText()),
-                        description = SafeReadString(() => opt.Description?.GetFormattedText()),
+                        title = SafeReadString(() => EventOptionLocalization.Format(
+                            opt.Title,
+                            locString => eventModel.DynamicVars.AddTo(locString),
+                            locString => locString.GetFormattedText())),
```
[the remark this answers is not in the document]
Comment (XenoAmess) [reply STS2AIAgent/Game/GameStateService.cs:3977]: Fixed in 8d949ba613b0f8f30ff69e058acaffa58298c7a2. `BuildEventOptionSignature` now formats both title and description through `EventOptionLocalization.Format`, injecting `eventModel.DynamicVars` before `GetFormattedText()`. I also added the `EventOptionLocalization.Signature` regression test, covering dynamic variables in both signature fields.

Validation with .NET SDK 9.0.317:
- `C:\Users\xenoa\AppData\Local\Microsoft\dotnet\dotnet.exe run --project STS2AIAgent.Tests/STS2AIAgent.Tests.csproj` — passed all 51 tests, including `EventOptionLocalization.DynamicVars`, `.Null`, and `.Signature`.
- `C:\Users\xenoa\AppData\Local\Microsoft\dotnet\dotnet.exe build STS2AIAgent/STS2AIAgent.csproj --no-restore` — could not complete because the external game assemblies `sts2.dll`, `0Harmony.dll`, and `GodotSharp.dll` were unavailable at the configured `Sts2DataDir`; the resulting errors are missing game/Godot types rather than test failures.
- `git diff --check` — passed.
